int32 senderID
int64[] stateIDs
int64[] succeededEps